tfhe/src/c_api/high_level_api/integers.rs

```diff
@@ -524,28 +581,28 @@ pub unsafe extern "C" fn compact_fhe_uint256_list_try_encrypt_with_compact_publi
         let values = slc
             .iter()
             .copied()
-            .map(crate::integer::U256::from)
+            .map(crate::integer::I256::from)
             .collect::<Vec<_>>();
         let inner =
-            <crate::high_level_api::CompactFheUint256List>::try_encrypt(&values, &public_key.0)
+            <crate::high_level_api::CompactFheInt256List>::try_encrypt(&values, &public_key.0)
                 .unwrap();
 
-        *result = Box::into_raw(Box::new(CompactFheUint256List(inner)));
+        *result = Box::into_raw(Box::new(CompactFheInt256List(inner)));
     })
 }
 
 #[no_mangle]
-pub unsafe extern "C" fn fhe_uint256_decrypt(
-    encrypted_value: *const FheUint256,
+pub unsafe extern "C" fn fhe_int256_decrypt(
+    encrypted_value: *const FheInt256,
```

Comment:
Why do the large ints require a manual management in that case ? I'm guessing the traits we have don't allow to do this ?

Reply:
Yes the trait used did not allow to use the macro for when the rust clear type is not the same as C clear type.

I changed it so that it works for decrypt.

But for try_encrypt_compact_list its a bit more complex so I leave it as is